src/sgame/sg_bot_util.cpp

```diff
@@ -245,6 +245,63 @@ static int GetMaxEquipmentCost( gentity_t const* self )
 	return max_value;
 }
 
+// This function returns a percent value indicating how much
+// resupply action would be desirable.
+// The returned value depends on:
+// * distance between the caller and the closest supply building
+// * caller's skill
+//
+// NOTE: the distance modifier is the same as for BotGetHealScore
+//
+// TODO: use a navigation distance
+// TODO: add poison support
+// TODO: in some cases (goon's barbs) the weapon have infiniteAmmo
+//   but still a finished limited of ammo. This special case needs
+//   special care, *or* (and that's likely a better solution) to
+//   stop considering having infinite ammo.
+float BotGetResupplyScore( gentity_t *self )
+{
+	float dist = 0;
+	auto const& closestBuildings = self->botMind->closestBuildings;
+	playerState_t const& ps = self->client->ps;
+	team_t team = G_Team( self );
+	weapon_t wp = BG_PrimaryWeapon( ps.stats );
+	weaponAttributes_t const* weapon = BG_Weapon( wp );
+
+	// why check team? Because of goons.
+	if ( weapon->infiniteAmmo && team == TEAM_HUMANS )
```

Comment:
Just return here if the team is not humans? It doesn't seem to do anything useful for aliens of any type.

Reply:
adv goons and barbs. I want to get them able to use that routine at some point.